### PR TITLE
Spring todolist 3

### PR DESCRIPTION
**Pull request by tonyYuhaoYangMcrS**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 3-Authorization-II/protect-web-api/AppCreationScripts/createdApps.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<html><body><table>
```
Comment (derisen) [inline 3-Authorization-II/protect-web-api/AppCreationScripts/createdApps.html:1]: Make sure to remove this file (might also look into adding this to .gitignore)

File: 3-Authorization-II/protect-web-api/AppCreationScripts/Configure.ps1 [hunk 2]
```diff
@@ -13,7 +13,7 @@ param(
 
  In case you don't have Microsoft.Graph.Applications already installed, the script will automatically install it for the current user
  
- There are four ways to run this script. For more information, read the AppCreationScripts.md file in the same folder as this script.
+ There are two ways to run this script. For more information, read the AppCreationScripts.md file in the same folder as this script.
```
Comment (derisen) [inline 3-Authorization-II/protect-web-api/AppCreationScripts/Configure.ps1:16]: the `sample.json` that was used to generate these scripts should also be committed

File: 3-Authorization-II/protect-web-api/webapp/src/main/resources/templates/footer.html [hunk 3]
```diff
@@ -11,7 +11,7 @@
...
         </a>
-            &nbsp; Does this sample address your <strong>learning objective</strong>?
+            &nbsp; Does this sample meet address your <strong>learning objective</strong>?
```
Comment (derisen) [inline 3-Authorization-II/protect-web-api/webapp/src/main/resources/templates/footer.html:14]: I think this is missing the link to the survey

File: 3-Authorization-II/protect-web-api/README.md [hunk 4]
```diff
@@ -1,226 +1,230 @@
...
-   - Select the **Add a permission** button and then,
-   - Ensure that the **My APIs** tab is selected.
-   - In the list of APIs, select the API `java-spring-resource-api`.
-   - In the **Delegated permissions** section, select the **Access 'java-spring-resource-api'** in the list. Use the search box if necessary.
-   - Select the **Add permissions** button at the bottom.
-
-#### Configure the client app (java-spring-webapp-client) to use your app registration
+    1. In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `webapp`.
+    1. Under **Supported account types**, select **Accounts in this organizational directory only**
+    1. Select **Register** to create the application.
```
Comment (derisen) [inline 3-Authorization-II/protect-web-api/README.md:174]: Missing Redirect URI configuration

File: 3-Authorization-II/protect-web-api/README.md [hunk 5]
```diff
@@ -1,226 +1,230 @@
...
-| `CHANGELOG.md`                                                                      | List of changes to the sample.                                                              |
-| `CONTRIBUTING.md`                                                                   | Guidelines for contributing to the sample.                                                  |
-| `LICENSE`                                                                           | The license for the sample.                                                                 |
+This sample demonstrates a Java Spring Web application calling a Java Spring Web API that is secured using Azure AD.
+
+1. The client Java Spring Web application uses the [Microsoft.Identity.Web](https://aka.ms/microsoft-identity-web) to sign-in a user and obtain a JWT [Access Token](https://aka.ms/access-tokens) from **Azure AD**.
+1. The access token is used as a *bearer* token to authorize the user to call the Java Spring Web API protected by **Azure AD**.
+1. The service uses the [Microsoft.Identity.Web](https://aka.ms/microsoft-identity-web) to protect the Web api, check permissions and validate tokens.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/protect-web-api/README.md:36]: you need to make updates to code gen to print the right stuff here

**Comment by derisen** [comment]
error3

@tonyYuhaoYangMcrS probably need a generic error page on front-end for rejections from the API like these

File: 3-Authorization-II/protect-web-api/webapp/src/main/resources/application.yml [hunk 6]
```diff
@@ -24,3 +24,11 @@ app:
   api:
     base-address: http://localhost:8082
     date-endpoint: /api/date
```
Comment (derisen) [inline 3-Authorization-II/protect-web-api/webapp/src/main/resources/application.yml:26]: you probably want to add the other endpoint here

File: 3-Authorization-II/protect-web-api/README.md [hunk 7]
```diff
@@ -1,226 +1,230 @@
...
+1. Open a terminal or the integrated VSCode terminal.
+1. In the root directory as this project, run `cd resource-api`. 
+1. run `mvn clean compile spring-boot:run`.
 1. Open a terminal or the integrated VSCode terminal.
-1. In the same directory as this readme file, run `cd resource-api`. Then run `mvn clean compile spring-boot:run`.
-1. Open a second terminal tab or the another integrated VSCode terminal.
-1. In the same directory as this readme file, run `cd webapp`. Then run `mvn clean compile spring-boot:run`.
+1. In the root directory as this project, run `cd webapp`. 
+1. run `mvn clean compile spring-boot:run`.
 1. Open your browser and navigate to `http://localhost:8080`.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/protect-web-api/README.md:223]: this section needs love from code gen, refer to 4-1 readme.. like in 4-1 it prints as follows



### Step 4: Running the sample

From your shell or command line, execute the following commands:

```console
    cd 4-WebApp-Your-API\4-1-MyOrg\TodoListService
    dotnet run
```

Then, open a separate command terminal and run: 

```console
    cd 4-WebApp-Your-API\4-1-MyOrg\Client
    dotnet run
```

File: 3-Authorization-II/protect-web-api/README.md [hunk 8]
```diff
@@ -1,226 +1,230 @@
...
 
-![Experience](./ReadmeFiles/app.png)
-
 ## Explore the sample
 
+![Experience](./ReadmeFiles/app.png)
```
Comment (derisen) [inline 3-Authorization-II/protect-web-api/README.md:226]: > Experience

Screenshot here is dead

File: 3-Authorization-II/protect-web-api/README.md [hunk 9]
```diff
@@ -1,226 +1,230 @@
 ---
 page_type: sample
-languages:
+name: Enable your Java Spring Boot web app to sign in users with Azure Active Directory and call a protected Web api
+description: Enable your Java Spring Boot web app to sign in users in your Azure AD tenant and call a protected Web api protected with Spring Security
+- languages:
   - java
 products:
-  - azure
+    - java
```
Comment (derisen) [inline 3-Authorization-II/protect-web-api/README.md:8]: Need to double check if these products are part of the [taxonomy](https://review.learn.microsoft.com/en-us/help/platform/metadata-taxonomies?branch=main) (`java `and `spring-boot` are most probably not, you should add `msal-java`)

**Comment by derisen** [comment]
@tonyYuhaoYangMcrS we should probably get rid of the entire `/api/date` endpoint and related code

File: 3-Authorization-II/protect-web-api/resource-api/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapi/SampleController.java [hunk 10]
```diff
@@ -4,65 +4,194 @@
...
         return false;
         
-	}
+    }
+    
+    private void createIfNotCreated(OAuth2AuthenticatedPrincipal principal) {
+        if (TDL == null) {
+            this.TDL = new ToDoList();
+            TDL.add(new ToDoListItem(principal.getAttribute("oid"), "finish todo"));
+            TDL.add(new ToDoListItem(principal.getAttribute("oid"), "overworked, seek help"));
```
Comment (kalyankrishna1) [inline 3-Authorization-II/protect-web-api/resource-api/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapi/SampleController.java:192]: >overworked, seek help [](http://example.com/codeflow?start=69&length=21)

please use normal ones

File: 3-Authorization-II/3-App-CallAPI/daemon/src/main/resources/application.yml [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+                registration:
+                    azure:
+                        authorization-grant-type: client_credentials
+                        client-authentication-method: CLIENT_SECRET_POST
+                        client-id: Enter_Your_Client_ID_Here
+                        client-secret: Enter_Your_Client_Secret_Here
+                        scope: api://Enter_Your_WebAPI_Client_ID_Here/.default
+app:
+  api:
+    base-address: http://localhost:8082
```
Comment (derisen) [inline 3-Authorization-II/3-App-CallAPI/daemon/src/main/resources/application.yml:19]: If possible, we should list the API endpoints we call in this file, then use those in the source code

File: 3-Authorization-II/3-App-CallAPI/README.md [hunk 12]
```diff
@@ -0,0 +1,393 @@
...
+extensions:
+- services: ms-identity
+- platform: Java
+- endpoint: AAD v2.0
+- level: 200
+- client: Java Spring Daemon
+- service: Java Spring Web API
+---
+
+# Enable your Java Spring Boot daemon to sign in users with Azure Active Directory and call a protected Web api
```
Comment (derisen) [inline 3-Authorization-II/3-App-CallAPI/README.md:20]: > # Enable your Java Spring Boot daemon to sign in users with Azure Active Directory and call a protected Web api

Daemon app does not sign in users. Should probably update this in sample.json and then generate again

File: 3-Authorization-II/3-App-CallAPI/README.md [hunk 13]
```diff
@@ -0,0 +1,393 @@
...
+## Overview
+
+This sample demonstrates a Java Spring Daemon calling a Java Spring Web API that is secured using Azure AD.
+
+## Scenario
+
+This sample demonstrates a Java Spring Daemon calling a Java Spring Web API that is secured using Azure AD.
+
+1. The client Java Spring Daemon uses the [MSAL Java](https://aka.ms/identityplatform) to sign-in a user and obtain a JWT [ID Token](https://aka.ms/id-tokens) and an [Access Token](https://aka.ms/access-tokens) from **Azure AD**.
+1. The **access token** is used as a *bearer* token to authorize the user to call the Java Spring Web API protected by **Azure AD**.
```
Comment (derisen) [inline 3-Authorization-II/3-App-CallAPI/README.md:44]: > The **access token** is used as a *bearer* token to authorize the user to call the Java Spring Web API protected by **Azure AD**.

Codegen should have generated this according to a daemon app (i.e. no mention of user). If that's not happening by default, can we create an issue in the Codegen repo to address this?

File: 3-Authorization-II/3-App-CallAPI/README.md [hunk 14]
```diff
@@ -0,0 +1,393 @@
...
+![Scenario Image](./AppCreationScripts/ReadmeFiles/topology.png)
+
+## Prerequisites
+
+* [JDK Version 15](https://jdk.java.net/15/). This sample has been developed on a system with Java 15 but may be compatible with other versions.
+* [Maven 3](https://maven.apache.org/download.cgi)
+* [Java Extension Pack for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) is recommended for running this sample in VSCode.
+* An **Azure AD** tenant. For more information, see: [How to get an Azure AD tenant](https://docs.microsoft.com/azure/active-directory/develop/test-setup-environment#get-a-test-tenant)
+* A user account in your **Azure AD** tenant.
+>This sample will not work with a **personal Microsoft account**. If you're signed in to the [Azure portal](https://portal.azure.com) with a personal Microsoft account and have not created a user account in your directory before, you will need to create one before proceeding.
```
Comment (derisen) [inline 3-Authorization-II/3-App-CallAPI/README.md:55]: > This sample will not work with a **personal Microsoft account**. If you're signed in to the [Azure portal](https://portal.azure.com) with a personal Microsoft account and have not created a user account in your directory before, you will need to create one before proceeding.

Same here: codegen shouldn't generate this if its a daemon app

File: 3-Authorization-II/3-App-CallAPI/AppCreationScripts/createdApps.html [hunk 15]
```diff
@@ -0,0 +1,5 @@
+<html><body><table>
```
Comment (derisen) [inline 3-Authorization-II/3-App-CallAPI/AppCreationScripts/createdApps.html:1]: reminder to not check-in these files

File: 3-Authorization-II/3-App-CallAPI/AppCreationScripts/README_STEPS.md [hunk 16]
```diff
@@ -0,0 +1,393 @@
+---
```
Comment (derisen) [inline 3-Authorization-II/3-App-CallAPI/AppCreationScripts/README_STEPS.md:1]: reminder to not check-in these files

File: 3-Authorization-II/protect-web-api/resource-api/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapi/SampleController.java (hunk 10 above)
Comment (tonyYuhaoYangMcrS) [reply 3-Authorization-II/protect-web-api/resource-api/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapi/SampleController.java]: changed to normal ones

File: 3-Authorization-II/3-App-CallAPI/README.md [hunk 17]
```diff
@@ -0,0 +1,393 @@
...
+
+### Step 4: Running the sample
+
+1. Open a terminal or the integrated VSCode terminal.
+1. In the root directory as this project, run `cd 3-Authorization-II\3-App-CallAPI\resource-api`.
+1. run `mvn clean compile spring-boot:run`.
+1. Open a terminal or the integrated VSCode terminal.
+1. In the root directory as this project, run `cd 3-Authorization-II\3-App-CallAPI\daemon`.
+1. run `mvn clean compile spring-boot:run`.
+1. Open your browser and navigate to `http://localhost:8080`.
```
Comment (salman90) [inline 3-Authorization-II/3-App-CallAPI/README.md:225]: We don't use the browser in this case.

File: 3-Authorization-II/3-App-CallAPI/README.md [hunk 18]
```diff
@@ -0,0 +1,393 @@
...
+This sample demonstrates a Java Spring Daemon calling a Java Spring Web API that is secured using Azure AD.
+
+## Scenario
+
+This sample demonstrates a Java Spring Daemon calling a Java Spring Web API that is secured using Azure AD.
+
+1. The client Java Spring Daemon uses the [MSAL Java](https://aka.ms/identityplatform) to sign-in a user and obtain a JWT [ID Token](https://aka.ms/id-tokens) and an [Access Token](https://aka.ms/access-tokens) from **Azure AD**.
+1. The **access token** is used as a *bearer* token to authorize the user to call the Java Spring Web API protected by **Azure AD**.
+
+![Scenario Image](./AppCreationScripts/ReadmeFiles/topology.png)
```
Comment (salman90) [inline 3-Authorization-II/3-App-CallAPI/README.md:46]: The image is not relevant to the scenario. please update the image.

File: 3-Authorization-II/3-App-CallAPI/README.md [hunk 19]
```diff
@@ -0,0 +1,393 @@
...
+	 * Add client Id of your client apps to allowedClientApps
+	 */
+	String[] allowedClientApps = new String[] {""};
+	for (int i = 0; i < allowedClientApps.length; i++ ) {
+		validators = AADHelpers.AADExtendedValidators(validators, allowedClientApps[i]);
+	}
+```	
+## How to deploy this sample to Azure
+
+<details>
```
Comment (salman90) [inline 3-Authorization-II/3-App-CallAPI/README.md:301]: The rest of the readme is collapsed under this title. Please check the spacing here.

File: 3-Authorization-II/3-App-CallAPI/daemon/src/main/java/com/microsoft/azuresamples/msal4j/configuration/OAuthClientConfiguration.java [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+
+
+@Configuration
+public class OAuthClientConfiguration {
+	
+    @Bean
```
Comment (derisen) [inline 3-Authorization-II/3-App-CallAPI/daemon/src/main/java/com/microsoft/azuresamples/msal4j/configuration/OAuthClientConfiguration.java:28]: We should thoroughly comment authN/authZ code like these, with links to the protocol that this is based on (e.g. https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow)

File: 3-Authorization-II/2-User-CallAPI/AppCreationScripts/README_STEPS.md [hunk 21]
```diff
@@ -1,26 +1,25 @@
 ---
 page_type: sample
```
Comment (kalyankrishna1) [inline 3-Authorization-II/2-User-CallAPI/AppCreationScripts/README_STEPS.md:2]: >page_type [](http://example.com/codeflow?start=0&length=9)

delete this

File: 3-Authorization-II/3-App-CallAPI/daemon/src/main/java/com/microsoft/azuresamples/msal4j/MsIdentitySpringBootDaemonApplication.java [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+    @Override
+    public void run(String... args) throws Exception{
+    	getTable();
+    	add("Testing");
+    	delete("6");
+    	edit("3", "Changed for app");
+    	details("2");
+    }
+    
+    public void getTable() throws Exception{
```
Comment (kalyankrishna1) [inline 3-Authorization-II/3-App-CallAPI/daemon/src/main/java/com/microsoft/azuresamples/msal4j/MsIdentitySpringBootDaemonApplication.java:51]: >getTable [](http://example.com/codeflow?start=16&length=8)

doesn't Java has a concept of code comments

File: 3-Authorization-II/3-App-CallAPI/README.md [hunk 23]
```diff
@@ -0,0 +1,393 @@
...
+* [Explore the sample](#explore-the-sample)
+* [Troubleshooting](#troubleshooting)
+* [How to deploy this sample to Azure](#how-to-deploy-this-sample-to-azure)
+* [Next Steps](#next-steps)
+* [Contributing](#contributing)
+* [Learn More](#learn-more)
+
+## Overview
+
+This sample demonstrates a Java Spring Daemon calling a Java Spring Web API that is secured using Azure AD.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/3-App-CallAPI/README.md:37]: > a Java Spring Daemon calling a Java  [](http://example.com/codeflow?start=24&length=37)

where is MSAL Java being used?

File: 3-Authorization-II/2-User-CallAPI/README.md [hunk 24]
```diff
@@ -0,0 +1,489 @@
...
+
+## Overview
+
+This sample demonstrates a Java Spring Web application calling a Java Spring Web API that is secured using Azure AD.
+
+## Scenario
+
+This sample demonstrates a Java Spring Web application calling a Java Spring Web API that is secured using Azure AD.
+
+1. The client Java Spring Web application uses the [Microsoft.Identity.Web](https://aka.ms/microsoft-identity-web) to sign-in a user and obtain a JWT [Access Token](https://aka.ms/access-tokens) from **Azure AD**.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/2-User-CallAPI/README.md:34]: > [Microsoft.Identity.Web](https://aka.ms/microsoft-identity-web) [](http://example.com/codeflow?start=50&length=64)

Where did this come from??

**Pull request closed without merging**